gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/ComputeManifestsUDF.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.iceberg;
+
+import com.esotericsoftware.minlog.Log;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.spark.sql.api.java.UDF1;
+
+class ComputeManifestsUDF implements UDF1<String, List<String>> {
+
+  private final FileIO io;
+
+  public ComputeManifestsUDF(FileIO io) {
+    this.io = io;
+  }
+
+  @Override
+  public List<String> call(String metadataLocation) {
+    TableMetadata metadata = TableMetadataParser.read(io, metadataLocation);
+    List<String> manifestFiles = new ArrayList<>();
+    metadata
+        .snapshots()
+        .forEach(
+            snapshot -> {
+              try {
+                manifestFiles.addAll(
+                    snapshot.allManifests().stream()
```

Comment:
The intention is to log the warning and move on instead of failing the whole expiry contents. 
I am catching only the `NotFoundException`, are you expecting to catch IO exceptions aswell ?

I will catch `UncheckedIOException` (RuntimeIOException) which can be thrown from Iceberg.